### PR TITLE
 fix(TDI-38396): fix null snowflake account

### PR DESCRIPTION
**Pull request by TotalP**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
jdbc:snowflake://null.snowflakecomputing.com/?null returned  as a result of SnowflakeConnectionProperties.getConnectionUrl() when account is null

**What is the new behavior?**
Specific error message provided

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://jira.talendforge.org/browse/TDI-38396

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/11/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/11/)

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java [hunk 1]
```diff
@@ -218,33 +218,44 @@ public Properties getJdbcProperties() {
     }
 
     public String getConnectionUrl() {
-        String queryString = "";
+        StringBuilder stringBuilder = new StringBuilder();
         String account = this.account.getStringValue();
 
+        if (account == null || account.isEmpty()) {
+            throw new IllegalArgumentException(" Missing user name");
```
Comment (chmyga) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java]: account is not a user name. User name is used during connection in properties.

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java [hunk 2]
```diff
@@ -218,33 +218,44 @@ public Properties getJdbcProperties() {
...
         String warehouse = this.warehouse.getStringValue();
         String db = this.db.getStringValue();
         String schema = this.schemaName.getStringValue();
-
         String role = this.role.getStringValue();
         String tracing = this.tracing.getStringValue();
 
-        if (null != warehouse && !"".equals(warehouse)) {
-            queryString = queryString + "warehouse=" + warehouse;
+        if (warehouse != null && !warehouse.isEmpty()) {
```
Comment (chmyga) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java]: why don't you move this part to appendProperty() method? it is used in every block of code where the appendProperty() method is called.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/26/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/26/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/53/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/53/)

**Pull request closed without merging**